File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    unordered_map<string, int> search_map;
    
    for(int i = 0; i < want.size(); i++) search_map.insert({want[i], i});
    
    for(int i = 0; i <= (discount.size()-10); i++){
        vector<int> possible(want.size(), 0);
        for(int j = i; j < i+10; j++){
            auto temp = search_map.find(discount[j]);
            if(temp!=search_map.end()) possible[temp->second]++;
        }
        if(possible==number) answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter
def solution(want, number, discount):
    answer = 0

    buy = {want[i]: number[i] for i in range(0, len(want))}

    for i in range(0, len(discount) - 9):
        counter = Counter(discount[i:i+10])

        for k, v in buy.items():
            if(k not in counter or v > counter[k]):
                break
        else:
            answer += 1

    return answer